src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -1065,6 +1085,207 @@ def _validate_up_args(cmd, source, artifact, build_env_vars, image, repo, regist
+You can then deploy your personalized template image by running the following:
+az containerapp up --name <app_name> --image <your_image> --model-name <model_name> --model-version <model_version> --model-registry <model_registry>
+Currently supported model list: https://aka.ms/aca/serverless-gpu-regions"""
+            )
+    model_asset_id, model_reference_endpoint, model_type, model_load_class = _validate_azml_model_existence(cmd, model_registry, model_name, model_version)
+    return model_asset_id, model_reference_endpoint, model_type, model_load_class, image_name
+
+
+def _set_azml_env_vars(cmd, cli_passed_env_vars, model_asset_id, model_reference_endpoint, model_format, model_load_classes, is_azml_mcr_app):
+    # We will automatically manage the required env vars for cx if the azml app is using mcr image.
+    # If the user is using a custom image, we will only manage the essential ones.
+    essential_env_vars = [f"AZURE_ML_MODEL_TYPE={model_format.lower()}", f"AZURE_ML_MODEL_ID={model_asset_id}", f"AZURE_ML_MODEL_PATH={model_reference_endpoint}"]
+    if is_azml_mcr_app and model_format.lower() != "custom" and (model_load_classes is None or model_load_classes == []):
+        raise ValidationError("""Model is currently not fully supported by our MCR image.
+Please visit https://github.com/microsoft/azure-container-apps/tree/main/templates to download and modify the container template to get best experience.
+You can then build and deploy the modified template from local path using --local-path option or use your own docker image.
+Currently supported model list: https://aka.ms/something""")
+    required_env_vars = essential_env_vars + model_load_classes if is_azml_mcr_app else essential_env_vars
+    new_env_vars = required_env_vars
+    if cli_passed_env_vars is not None:
+        for cli_env_var in cli_passed_env_vars:
+            key, _ = cli_env_var.split("=")
+            if key not in required_env_vars:
+                new_env_vars.append(cli_env_var)
+    return new_env_vars
+
+
+def _validate_azml_env_and_create_if_needed(cmd, app, env, cli_input_environment_name, resource_group, cli_input_resource_group_name, workload_profile_name):
+    cpu = None
+    memory = None
+    if app.check_exists():
+        wp_name = app.get()["properties"]["workloadProfileName"]
+        env_name = app.get()["properties"]["environmentId"].split("/")[-1]
+        env_rg_name = parse_resource_id(app.get()["properties"]["environmentId"])["resource_group"]
+        if cli_input_environment_name and cli_input_environment_name.lower() != env_name.lower():
+            raise ValidationError(f"{app.name} in resource group {app.resource_group.name} found under {env_name}, which is different from the one specified in --environment-name {cli_input_environment_name}. Please specify the correct environment.")
```

Comment:
The name of the params is not correct, please check

```suggestion
            raise ValidationError(f"{app.name} in resource group {app.resource_group.name} found under {env_name}, which is different from the one specified in --environment {cli_input_environment_name}. Please specify the correct environment.")
```